applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -542,6 +551,41 @@ impl wallet_server::Wallet for WalletGrpcServer {
         Ok(Response::new(GetTransactionInfoResponse { transactions }))
     }
 
+    async fn stream_transaction_events(
+        &self,
+        _request: tonic::Request<TransactionEventRequest>,
+    ) -> Result<Response<Self::StreamTransactionEventsStream>, Status> {
+        let (mut sender, receiver) = mpsc::channel(100);
+        task::spawn(async move {
+            let event_listener = &WALLET_EVENT_LISTENER.receiver;
+            loop {
+                let transaction_event = match event_listener.recv_timeout(Duration::from_secs(1)) {
+                    Ok(source) => TransactionEvent::try_from(source),
+                    Err(_) => continue,
+                };
+                if transaction_event.is_ok() {
+                    let response = TransactionEventResponse {
+                        transaction: Some(transaction_event.unwrap()),
+                    };
+                    match sender.send(Ok(response)).await {
+                        Ok(_) => (),
+                        Err(err) => {
+                            warn!(target: LOG_TARGET, "Error sending transaction via GRPC:  {}", err);
+                            match sender.send(Err(Status::unknown("Error sending data"))).await {
```

Comment:
You've just failed to send a message. Now you're trying to send the error status to the very thing that just failed.

Is this intentional?

Reply:
We do this in other places in other GRPC with streams. At worst case, you fail twice, at best, you get the error to the client. I think in most cases it ends up being a double fail